syncapi/synctypes/clientevent.go

```diff
@@ -154,60 +155,276 @@ func ToClientEvent(se gomatrixserverlib.PDU, format ClientEventFormat, sender st
 
 	if format != FormatSyncFederation {
 		if se.Version() == gomatrixserverlib.RoomVersionPseudoIDs {
-			ce.SenderKey = se.SenderID()
+			err := updatePseudoIDs(&ce, se, userIDForSender, format)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	return ce
+
+	return &ce, nil
 }
 
-// ToClientEvent converts a single server event to a client event.
-// It provides default logic for event.SenderID & event.StateKey -> userID conversions.
-func ToClientEventDefault(userIDQuery spec.UserIDForSender, event gomatrixserverlib.PDU) ClientEvent {
-	sender := spec.UserID{}
-	validRoomID, err := spec.NewRoomID(event.RoomID())
+func updatePseudoIDs(ce *ClientEvent, se gomatrixserverlib.PDU, userIDForSender spec.UserIDForSender, format ClientEventFormat) error {
+	ce.SenderKey = se.SenderID()
+
+	validRoomID, err := spec.NewRoomID(se.RoomID())
```

Comment:
This has somewhat bothered me all the time.
We have, in theory, already a valid event, why do we need to parse the roomID it again as a `RoomID`.. `RoomID` in the `eventFields` struct should probably be a `spec.RoomID`. (at some point :D)

Reply:
Done! :)